EmployeeSalary.Api/V1/Services/SalaryCalculationService.cs

```diff
@@ -0,0 +1,27 @@
+﻿using AutoMapper;
+using EmployeeSalary.Api.Models.Entities;
+using EmployeeSalary.Api.Models.Shards;
+using EmployeeSalary.Api.V1.DTOs;
+
+namespace EmployeeSalary.Api.V1.Services
+{
+    public class SalaryCalculationService : ISalaryCalculationService
+    {
+        private readonly IMapper _mapper;
+        public SalaryCalculationService(IMapper mapper)
+        {
+            _mapper = mapper;
+        }
+        public double CalculateSalary<TInputDto, TEmployee>(TInputDto input) 
+            where TEmployee: Employee
+            where TInputDto : BasicSalaryCalculationDto
+        {
+            var employee = _mapper.Map<TEmployee>(input);
```

Comment:
no need for auto mapper, just passing the type to employee.CalcuateSalary should be enough, you can check for null there. also, you're using validation attributes. this will never be null. so even null checks are not needed.

Reply:
What do you mean by passing the type to (employee.CalcuateSalary), because as you know without autoMApper we can't get the object that is called employee, that contains the CalculateSalary()
![image](https://user-images.githubusercontent.com/62394795/185785865-13bf269d-e3d5-477e-ad2b-75a643f31a86.png)